pkg/bootstrap/docker/dockerhelper/filetransfer.go

```diff
@@ -116,18 +117,16 @@ func DownloadDirFromContainer(client *docker.Client, container, src, dst string)
 	defer downloader.Close()
 	tarReader := &removeLeadingDirectoryAdapter{Reader: tar.NewReader(downloader)}
 
-	t := stitar.New()
+	t := s2itar.New(s2iutil.NewFileSystem())
 	return t.ExtractTarStreamFromTarReader(dst, tarReader, nil)
 }
 
 // UploadFileToContainer uploads a file to a remote container.
 func UploadFileToContainer(client *docker.Client, container, src, dest string) error {
 	uploader, errch := newContainerUploader(client, container, filepath.Dir(dest))
 
-	nullWalkFunc := func(path string, info os.FileInfo, err error) error { return err }
-
-	t := stitar.New()
-	err := t.StreamFileAsTarWithCallback(src, filepath.Base(dest), uploader, nullWalkFunc, false)
+	t := s2itar.New(s2iutil.NewFileSystem())
+	err := t.CreateTarStream(src, false, uploader)
```

Comment:
it's the best I could think of - tarWriter.Close() sends the terminating '\0's record file which are supposed to indicate to tar that its stream isn't erroneously truncated.